casper/src/test/scala/coop/rchain/casper/RholangBuildTest.scala

```diff
@@ -60,4 +74,40 @@ class RholangBuildTest extends FlatSpec with Matchers {
       }
   }
 
+  "Our build system" should "execute the genesis block" in effectTest {
+    HashSetCasperTestNode
+      .standaloneEff(genesis)
+      .use { node =>
+        val vaults = Source
+          .fromResource("wallets.txt")
+          .getLines()
+          .map { line =>
+            line.split(",") match {
+              case Array(ethAddress, initRevBalanceStr, _) =>
+                Vault(RevAddress.fromEthAddress(ethAddress), initRevBalanceStr.toLong)
+            }
+          }
+          .toSeq
+        val genesisPk = defaultPub
+
+        import node._
+
+        for {
+          deploy <- ConstructDeploy.sourceDeployNowF(
```

Comment:
Here you're re-deploying `RevGenerator`. If the time of this deploy, using the changes from the parallelization PR I opened, is what you're measuring, then that explains the performance disparity. I'm measuring the time to compute the genesis block with 16K vaults. You're measuring the time to compute `RevGenerator` with 16K vaults.

Reply:
It shouldn't make a difference.